Assets/LeapMotionModules/UI/Interaction/InteractionBehaviour.cs

```diff
@@ -0,0 +1,692 @@
+
+    [Header("Advanced Settings")]
+    [Tooltip("Warping manipulates the graphical (but not physical) position of grasped objects "
+           + "based on the movement of the Leap hand so the objects appear to move with less latency.")]
+    public bool graspHoldWarpingEnabled__curIgnored = true; // TODO: Warping not yet implemented.
+
+    protected override void Awake() {
+      base.Awake();
+      rigidbodyWarper = new RigidbodyWarper(interactionManager, this.transform, _body, 0.25F);
+    }
+
+    protected override void Start() {
+      base.Start();
+
+      RefreshPositionLockedState();
+      InitGrasping();
+
+      InitInternal();
+    }
+
+    /// <summary>
+    /// InteractionManager manually calls this directly
+    /// after all InteractionHands are updated (in FixedUpdate).
+    /// 
+    /// Hovering uses its update to provide per-object (as opposed to per-hand)
+    /// hover callbacks, e.g. OnObjectHoverStay(), which fires once per frame
+    /// while the object is hovered by any number of hands greater than zero.
+    /// 
+    /// Grasping uses its update to provide per-object (as opposed to per-hand)
+    /// grasp callbacks, e.g. OnObjectGraspBegin(), similarly. It will also
+    /// fire multi-grasp callbacks, e.g. OnMultiGraspBegin(), but only if
+    /// multiHandGrasping is enabled.
+    /// </summary>
+    public override void FixedUpdateObject() {
+      FixedUpdateHovering();
+      FixedUpdatePrimaryHovering();
+      FixedUpdateContact();
+      FixedUpdateGrasping();
+
+      FixedUpdateCollisionMode();
+    }
+
+    // TODO: Currently this gets the distance from the point to this transform, but this will
+    // need to incorporate distance to the rigidbody.
+    private float GetInteractionDistanceToPoint(Vector3 point) {
+      return Vector3.Distance(point, this.transform.position);
+    }
+
+    #region Hovering
+
+    // Logistics for providing per-object (instead of per-hand) Hover callbacks.
+    private Hand  _closestHoveringHand = null;
+    private float _closestHoveringHandDistance = float.PositiveInfinity;
+    private Hand  _closestJustStoppedHoveringHand = null; // Provided for OnObjectHoverEnd.
+    private int   _hoveringHandsCountLastFrame = 0;
+    private int   _hoveringHandsCount = 0;
+
+    private Hand _publicClosestHoveringHand = null;
+
+    // Runs after InteractionHands have done FixedUpdateHand.
+    private void FixedUpdateHovering() {
+      if (_hoveringHandsCount > 0) {
+        if (_hoveringHandsCountLastFrame == 0) {
+          _publicClosestHoveringHand = _closestHoveringHand;
+          OnObjectHoverBegin(_closestHoveringHand);
+        }
+        else {
+          _publicClosestHoveringHand = _closestHoveringHand;
+          OnObjectHoverStay(_closestHoveringHand);
+        }
+      }
+      else if (_hoveringHandsCountLastFrame > 0) {
+        _closestHoveringHand = null;
+        OnObjectHoverEnd(_closestJustStoppedHoveringHand);
```

Comment:
Another general lifecycle discussion.  I would argue that Stay should be dispatched after all Begin/End callbacks are dispatched.